shared/src/address/payload.rs

```diff
@@ -151,6 +151,17 @@ impl From<&Payload> for Protocol {
     }
 }
 
+impl TryFrom<Payload> for DelegatedAddress {
+    type Error = ();
+
+    fn try_from(value: Payload) -> Result<Self, Self::Error> {
+        match value {
+            Payload::Delegated(d) => Ok(d),
+            _ => Err(()),
```

Comment:
We should probably do what we do for ID addresses and introduce a new `Error::NonDelegatedAddress` error. It's not great, but it's better than unit.